src/votekit/elections/election_types.py

```diff
@@ -1117,3 +1120,317 @@ def __init__(
+                weights = [b.weight for b in ballots]
+                random_ballot = random.choices(
+                    self.state.profile.ballots, weights=weights, k=1
+                )[0]
+                # randomly choose a winner according to first place rankings
+                winning_candidate = random.choice(list(random_ballot.ranking[0]))
+
+            # some formatting to make it compatible with ElectionState, which
+            # requires a list of sets of strings
+            elected = [{winning_candidate}]
+
+            # remove the winner from the ballots
+            new_ballots = remove_cand(winning_candidate, self.state.profile.ballots)
+            new_profile = PreferenceProfile(ballots=new_ballots)
+
+            # determine who remains
+            remaining = [{c} for c in remaining if c != winning_candidate]
+
+            # update for the next round
+            self.state = ElectionState(
+                curr_round=self.state.curr_round + 1,
+                elected=elected,
+                eliminated_cands=[],
+                remaining=remaining,
+                profile=new_profile,
+                previous=self.state,
+            )
+
+            # if this is the last round, move remaining to eliminated
+            if not self.next_round():
+                self.state = ElectionState(
+                    curr_round=self.state.curr_round,
+                    elected=elected,
+                    eliminated_cands=remaining,
+                    remaining=[],
+                    profile=new_profile,
+                    previous=self.state.previous,
+                )
+            return self.state
+
+    def run_election(self):
+        # run steps until we elect the required number of candidates
+        while self.next_round():
+            self.run_step()
+
+        return self.state
+
+
+class PluralityVeto(Election):
+    """
+    Scores each candidate by their plurality (number of first) place votes,
+    then in a randomized order it lets each voter decrement the score of their
+    least favorite candidate. The candidate with the largest score at the end is
+    then chosen as the winner.
+
+    Args:
+      profile (PreferenceProfile): PreferenceProfile to run election on.
+      seats (int): Number of seats to elect.
+
+    Attributes:
+      _profile (PreferenceProfile): PreferenceProfile to run election on.
+      seats (int): Number of seats to be elected.
+
+    """
+
+    def __init__(self, profile: PreferenceProfile, seats: int):
+        super().__init__(profile, ballot_ties=False)
+        self.seats = seats
+
+    def next_round(self) -> bool:
+        """
+        Determines if another round is needed.
+
+        Returns:
+            (bool) True if number of seats has not been met, False otherwise.
+        """
+        cands_elected = [len(s) for s in self.state.winners()]
+        return sum(cands_elected) < self.seats
+
+    def run_step(self):
```

Comment:
Okay: let's refactor this. Each round should end when one candidate is eliminated.
Then remove them from the profile.  Start a new round.

You'll need to ensure that the random order is preserved. Should each ballot have weight 1? Yes: de-coupling the ballots with weight bigger than 1. In the init method, validate that each weight is an integer.